src/ddmd/mars.d

```diff
@@ -1498,6 +1493,33 @@ private bool parseCommandLine(const ref Strings arguments, const size_t argc, re
         errors = true;
     }
 
+    /************************************
+     * Convert string to integer.
+     * Params:
+     *  p = pointer to start of string digits, ending with 0
+     *  max = max allowable value (inclusive)
+     * Returns:
+     *  uint.max on error, otherwise converted integer
+     */
+    static pure uint parseDigits(const(char)*p, const uint max)
+    {
+        uint value;
+        bool overflow;
+        while (1)
+        {
+            const c = *p;
```

Comment:
Save a comparison per pass, simplify flow:

```d
for (uint d; (d = uint(*p) - uint('0')) <= 9; ++p)
{
    import core.checkedint : mulu, addu;
    value = mulu(value, 10, overflow);
    value = addu(value, d, overflow);
}
return (overflow || value > max || c) ? uint.max : value;
```

Reply:
It's beautiful, but that `c` in the return expression is a problem. `d` has gone out of scope, so I'd have to use `*p`, meaning an extra indirection.

Reply:
Ok, I made the change.

Reply:
That's only one indirection, not one per loop. You can of course change the code like this at the cost of some minor obfuscation:

```d
uint d;
for (; (d = uint(*p) - uint('0')) <= 9; ++p)
{
    import core.checkedint : mulu, addu;
    value = mulu(value, 10, overflow);
    value = addu(value, d, overflow);
}
return (overflow || value > max || d + '0' != 0) ? uint.max : value;
```

Reply:
Eh, I realized there's indeed a better version with an infinite loop:

```d
while (1)
{
    const d = uint(*p) - uint('0');
    if (d > 9)
        return uint.max;
    import core.checkedint : mulu, addu;
    value = mulu(value, 10, overflow);
    value = addu(value, d, overflow);
    ++p;
}
return (overflow || value > max) ? uint.max : value;
```

Looks like this does minimal work.